shared/src/main/scala/mlscript/Typer.scala

```diff
@@ -62,15 +62,47 @@ class Typer(var dbg: Boolean, var verbose: Bool, var explainErrors: Bool, val ne
       env: MutMap[Str, TypeInfo],
       mthEnv: MutMap[(Str, Str) \/ (Opt[Str], Str), MethodType],
       lvl: Int,
+      qenv: MutMap[Str, SkolemTag], // * SkolemTag for variables in quasiquotes
+      fvars: MutSet[ST], // * Free variables
+      inQuote: Bool, // * Is in quasiquote
       inPattern: Bool,
+      funDefs: MutMap[Str, DelayedTypeInfo],
```

Comment:
Why is this needed? If the thing is in scope, its type info should already be accessible via the context.